src/vmm/src/devices/virtio/net/device.rs

```diff
@@ -1394,14 +1518,86 @@ pub mod tests {
             .check_used_elem(0, 0, frame_1.len().try_into().unwrap());
         th.rxq.dtable[0].check_data(&frame_1);
         th.rxq.dtable[1].check_data(&[0; 500]);
-        th.rxq.dtable[2].check_data(&[0; 526]);
+        th.rxq.dtable[2].check_data(&[0; MAX_BUFFER_SIZE - 1000]);
         // Check that the 2nd frame was written successfully to the 2nd Rx descriptor chain.
         header_set_num_buffers(frame_2.as_mut_slice(), 1);
         th.rxq
             .check_used_elem(1, 3, frame_2.len().try_into().unwrap());
         th.rxq.dtable[3].check_data(&frame_2);
         th.rxq.dtable[4].check_data(&[0; 500]);
-        th.rxq.dtable[2].check_data(&[0; 526]);
+        th.rxq.dtable[5].check_data(&[0; MAX_BUFFER_SIZE - 1000]);
+    }
+
+    #[test]
+    fn test_rx_multiple_frames() {
+        let mem = single_region_mem(3 * MAX_BUFFER_SIZE);
+        let th = TestHelper::get_default(&mem);
+        rx_multiple_frames(th);
+    }
+
+    #[test]
+    fn test_rx_multiple_frames_mrg() {
+        let mem = single_region_mem(3 * MAX_BUFFER_SIZE);
+        let mut th = TestHelper::get_default(&mem);
+        // VIRTIO_NET_F_MRG_RXBUF is not enabled by default
+        th.net().acked_features = 1 << VIRTIO_NET_F_MRG_RXBUF;
+        rx_multiple_frames(th);
+    }
+
+    fn rx_mrg_rxbuf_only(mut th: TestHelper) {
+        th.activate_net();
+
+        // Create 2 valid Rx avail descriptor chains. Each one is 32K so
```

Comment:
They are not 32K.

Reply:
this were remains at my initial attempts. Apparently, trying to sent big packet with `inject_tap_tx_frame` will fail.
Fixed now.